core/domain/user_services_test.py

```diff
@@ -2288,20 +2288,86 @@ def test_assign_user_review_question_suggestion(self):
+        all_reviewers = user_services.get_all_reviewers_contribution_rights()
         self.assertItemsEqual(
             [reviewer.id for reviewer in all_reviewers],
             [self.voice_artist_id, self.translator_id])
 
+    def test_get_reviewer_user_ids_to_notify_when_reviewers_want_notifications(
+            self):
+        # Assert that there are no reviewers at the start.
+        self.assertEqual(
+            user_services.get_all_reviewers_contribution_rights(), [])
+        # Add a question reviewer and a translation reviewer.
+        user_services.allow_user_to_review_question(self.question_reviewer_id)
+        user_services.allow_user_to_review_translation_in_language(
+            self.translator_id, 'hi')
+        # Ensure that these reviewers want email updates.
+        user_services.update_email_preferences(
+            self.question_reviewer_id, True,
+            feconf.DEFAULT_EDITOR_ROLE_EMAIL_PREFERENCE,
+            feconf.DEFAULT_FEEDBACK_MESSAGE_EMAIL_PREFERENCE,
+            feconf.DEFAULT_SUBSCRIPTION_EMAIL_PREFERENCE)
+        user_services.update_email_preferences(
+            self.translator_id, True,
+            feconf.DEFAULT_EDITOR_ROLE_EMAIL_PREFERENCE,
+            feconf.DEFAULT_FEEDBACK_MESSAGE_EMAIL_PREFERENCE,
+            feconf.DEFAULT_SUBSCRIPTION_EMAIL_PREFERENCE)
+
+        reviewer_ids_to_notify = user_services.get_reviewer_user_ids_to_notify(
+        )
+
+        self.assertEqual(len(reviewer_ids_to_notify), 2)
+        self.assertIn(self.question_reviewer_id, reviewer_ids_to_notify)
+        self.assertIn(self.translator_id, reviewer_ids_to_notify)
+
+    def test_get_reviewer_user_ids_to_notify_when_reviewers_do_not_want_emails(
+            self):
+        # Assert that there are no reviewers at the start.
+        self.assertEqual(
+            user_services.get_all_reviewers_contribution_rights(), [])
+        # Add a question reviewer and a translation reviewer.
+        user_services.allow_user_to_review_question(self.question_reviewer_id)
+        user_services.allow_user_to_review_translation_in_language(
+            self.translator_id, 'hi')
+        # Ensure that these reviewers do not want email updates.
+        user_services.update_email_preferences(
+            self.question_reviewer_id, False,
+            feconf.DEFAULT_EDITOR_ROLE_EMAIL_PREFERENCE,
+            feconf.DEFAULT_FEEDBACK_MESSAGE_EMAIL_PREFERENCE,
+            feconf.DEFAULT_SUBSCRIPTION_EMAIL_PREFERENCE)
+        user_services.update_email_preferences(
+            self.translator_id, False,
+            feconf.DEFAULT_EDITOR_ROLE_EMAIL_PREFERENCE,
+            feconf.DEFAULT_FEEDBACK_MESSAGE_EMAIL_PREFERENCE,
+            feconf.DEFAULT_SUBSCRIPTION_EMAIL_PREFERENCE)
+
+        reviewer_ids_to_notify = user_services.get_reviewer_user_ids_to_notify(
+        )
+
+        self.assertEqual(len(reviewer_ids_to_notify), 0)
+
+    def test_get_reviewer_user_ids_to_notify_when_there_are_no_reviewers(
```

Comment:
Here & above: when possible, test names should include the expected outcome, e.g.

``test_get_reviewer_user_ids_to_notify_returns_empty_for_no_reviewers``
``test_get_reviewer_user_ids_to_notify_returns_reviewers_do_not_want_emails``

Etc. Ditto for all test methods in this PR.

Reply:
Ah interesting, then I might have actually misunderstood the purpose of the second test. Perhaps:

``test_get_reviewer_user_ids_reviewers_dont_want_emails_returns_empty``

Would that work? I think it's under the character limit.

Usually there are ways to abbreviate. Feel free to ping me if you want to chat about specific test names--I have a lot of practice shortening them. :)

Reply:
That almost works but I feel like `test_get_reviewer_user_ids_to_notify_when_reviewers_do_not_want_emails` is still a bit clearer because your suggestion does not encompass the method name that I'm testing (it makes it seem more about reviewer user_ids and not reviewer user_ids to notify)

Reply:
Ack, thanks for considering the different options.